### PR TITLE
bro: 2.5.5 -> 2.6.2

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Security fix release mentioned at top of notes, FWIW:

https://raw.githubusercontent.com/zeek/zeek/bb979702cf9a2fa67b8d1a1c7f88d0b56c6af104/NEWS

----

Seems reasonable but if anyone actually familiar can test or otherwise
review that'd be appreciated!

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by tobim** [comment]
Initial Feedback:
A big change that comes with 2.6 is the introduction of the `broker` communications layer and library, this has some implications:
- Add `caf` to buildInputs and `"-DCAF_ROOT_DIR=${caf}"` to cmakeFlags, otherwise the internal version will be used adding unnecessary compilation time.
- Add `rocksdb` to buildInputs to enable brokers store module.
- The broker python bindings require the `ipaddress` module which was introduced in python-3.3, so the python parameter should be switched to python3.
- It would be nice to have a `toPythonModule` entry in `pythonPackages.nix`, otherwise the module will be hidden from users.
- The `geoip` dependency can be replaced with `libmaxminddb`, this also enables a requirement for `ncurses` in one of the tests.

**Comment by tobim** [comment]
Oh, and you need to add `openssl` to the buildInputs for `caf`.

**Comment by risicle** [comment]
BTW this builds on darwin if you enable the platform. Tested on macos 10.13.

File: pkgs/applications/networking/ids/bro/default.nix [hunk 1]
```diff
@@ -2,16 +2,20 @@
 , geoip, gperftools, python, swig }:
 
 stdenv.mkDerivation rec {
-  name = "bro-2.5.5";
+  name = "bro-2.6.2";
 
   src = fetchurl {
     url = "https://www.bro.org/downloads/${name}.tar.gz";
```
Comment (geistesk) [inline pkgs/applications/networking/ids/bro/default.nix]: With the renaming of the project, this address now forwards to the new domain. So perhaps the domain should already be updated.

```nix
url = "https://www.zeek.org/downloads/${name}.tar.gz";
```

File: pkgs/applications/networking/ids/bro/default.nix [hunk 2]
```diff
@@ -2,16 +2,20 @@
...
-    sha256 = "1kvkiq8jjsqryry9jd4vw45pbfb46jly988mq62mv4sd1fqsxwhq";
+    sha256 = "19n0xai1mndx2i28q9cnszam57r6p6zqhprxxfpxh7bv7xpqgxkd";
   };
 
   nativeBuildInputs = [ cmake flex bison file ];
   buildInputs = [ openssl libpcap perl zlib curl geoip gperftools python swig ];
 
+  # Indicate where to install the python bits, since it can't put them in the "usual"
+  # locations as those paths are read-only.
+  cmakeFlags = [ "-DPY_MOD_INSTALL_DIR=${placeholder "out"}/${python.sitePackages}" ];
```
Comment (geistesk) [inline pkgs/applications/networking/ids/bro/default.nix]: Most Zeek/Bro installation come with the *auxiliary tools*, like `bro-cut`. Thus, the `"-DINSTALL_AUX_TOOLS:BOOL=true"` cmake flag should be added.

**Review by oxzi — COMMENTED** [review]
I have tested this PR and it worked fine for me.

However, the project changed its name from Bro to Zeek and perhaps we should adopt this amendment. An alias for `bro` could be added for compatibility reasons.

**Comment by tobim** [comment]
The renaming from `Bro` to `Zeek` will be put into effect with version 3. So I think we should keep the name until then.

**Pull request closed without merging**